src/loader.mjs

```diff
@@ -33,7 +34,22 @@ const createLoader = () => {
     });
   };
 
-  return { loadFiles };
+  /**
+   * Loads the JavaScript source files and transforms them into VFiles
+   *
+   * @param {Array<string>} filePaths
+   */
+  const loadJsFiles = filePaths => {
+    filePaths = filePaths.filter(filePath => existsSync(filePath));
+
+    return filePaths.map(async filePath => {
+      const fileContents = await readFile(filePath, 'utf-8');
+
+      return new VFile({ path: filePath, value: fileContents });
```

Comment:
I am a bit concerned of the memory allocation that this will cause, some of the source files can be humungous. Have you made some comparisons of heap size differences?